### PR TITLE
User favorites endpoint

### PR DESCRIPTION
**Pull request by jorwoods**
Reopening #591. I closed accidentally instead of simply commenting.

### CODE REVIEW

**Comment by t8y8** [comment]
@shinchris I reviewed this before, a second opinion might be good (the old PR talks about why I requested a simple dict, I'm starting to reconsider but what do you think?)

**Comment by jorwoods** [comment]
Per our prior discussion, I like the dictionary approach as it separates the items out clearly, while still being iterable on the empty versions.

File: tableauserverclient/models/favorites_item.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import logging
+from .workbook_item import WorkbookItem
+from .view_item import ViewItem
+from .project_item import ProjectItem
+from .datasource_item import DatasourceItem
+
+logger = logging.getLogger('tableau.models.favorites_item')
+
+
+class Favorite:
```
[the remark this answers is not in the document]
Comment (jorwoods) [reply tableauserverclient/models/favorites_item.py]: - Renamed to FavoriteItem
- Moved enum inside FavoriteItem
- Renamed enum to Type

File: tableauserverclient/models/favorites_item.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Workbook = 'workbook'
+    Datasource = 'datasource'
+    View = 'view'
+    Project = 'project'
+
+
+def get_favorites(server_response, namespace):
```
[the remark this answers is not in the document]
Comment (jorwoods) [reply tableauserverclient/models/favorites_item.py]: - Moved get_favorites to within FavoriteItem class
- Renamed to from_response
- Changed from_response to be classmethod for api consistency

**Comment by jorwoods** [comment]
@shinchris Made the changes you suggested.

**Comment by shinchris** [comment]
Thanks @jorwoods for the PR!

**Pull request merged**